front/src/services/bedrock.ts

```diff
@@ -0,0 +1,175 @@
+import { toast } from "sonner";
+import { z } from "zod";
+
+import { AlephService } from "@/services/aleph";
+import { EncryptionService } from "@/services/encryption";
+
+export const FileEntrySchema = z.object({
+	path: z.string().regex(/^(\/[A-Za-z0-9-_+.]+)+$/, {
+		message: "Invalid URI format. It should start with '/', contain valid path segments and do not end with a '/'.",
+	}),
+	post_hash: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid hash format. It should be a 64-character hexadecimal string.",
+	}),
+});
+
+export const EncryptedFileEntrySchema = z.object({
+	path: z.string().regex(/^[a-f0-9]+$/, { message: "Invalid data, it should be encrypted and in hex format." }),
+	post_hash: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid hash format. It should be a 64-character hexadecimal string.",
+	}),
+});
+
+export const FileMetaSchema = z.object({
+	key: z.string().regex(/^[a-f0-9]{64}$/, {
+		message: "Invalid key format. It should be a 64-character hexadecimal string.",
+	}),
+	iv: z.string().regex(/^[a-f0-9]{32}$/, {
+		message: "Invalid IV format. It should be a 32-character hexadecimal string.",
+	}),
+	ipfs_hash: z.string().regex(/^[a-f0-9]{64}$/, {
```

Comment:
it's not the IPFS hash, it's the hash of the STORE message on Aleph (that handles the internal storage for us, which is inline on Aleph for small files and using IPFS for large files).
So I would call this `store_hash`